### DISCUSSION TITLE
There are various questions and posts about scan/associated_scan over ragged data, what about sparse arrays now and in the future?

### DISCUSSION THREAD

**Discussion by cottrell** [jax-ml/jax#25142] [Q&A] [answered]
I'm trying to guage what are the possibilities and plans in the future for jax regarding scans over sparse/ragged data. There is a common problem that comes up in any kind of filter with irregular observations. I have tried a a) sparse non-ragged arrays b) indices int dynamic slicing method etc but so far have hit blockers on all approaches.

It is fine to make other compromises/work-arounds for now but wondering what are the hard constraints on this stuff for the future.

**Comment by jakevdp** [comment, marked as answer]
> wondering what are the hard constraints on this stuff for the future.

The main hard constraint at the moment is that the XLA complier and StableHLO more generally do not support sparse or ragged data structures, which limits the ability of the JAX project to do anything non-experimental in this area.
